src/main/java/org/sonarlint/intellij/ui/SecurityHotspotsPanel.java

```diff
@@ -151,29 +152,43 @@ private void createFindingDetailsPanel() {
     findingDetailsPanel = new FindingDetailsPanel(project, this, FindingKind.SECURITY_HOTSPOT);
   }
 
-  public int updateFindings(Map<VirtualFile, Collection<LiveSecurityHotspot>> findings) {
-    currentFindings = findings;
+  public int updateHotspots(Map<VirtualFile, Collection<LiveSecurityHotspot>> hotspots) {
+    currentFindings = hotspots;
 
     if (project.isDisposed()) {
       return 0;
     }
 
     if (status instanceof Supported) {
-      securityHotspotCount = securityHotspotTreeBuilder.updateModelWithoutFileNode(findings, "No Security Hotspots found", true);
-      oldSecurityHotspotCount = oldSecurityHotspotTreeBuilder.updateModelWithoutFileNode(findings, "No Security Hotspots found", false);
+      if (getService(project, CleanAsYouCodeService.class).shouldFocusOnNewCode()) {
+        var oldHotspots = hotspots.entrySet().stream()
```

Comment:
We can extract this stream to a method that takes a boolean for forOldHotspots as a parameter